llm_studio/src/trl/trainer.py

```diff
@@ -651,15 +682,62 @@ def compute_rewards(
             rewards.append(reward)
         return torch.stack(rewards), torch.stack(non_score_rewards)
 
+    def _kl_penalty(
+        self, logprob: torch.FloatTensor, ref_logprob: torch.FloatTensor
+    ) -> torch.FloatTensor:
+        if self.cfg.training.full_kl_penalty:
+            # Flip is required due to this issue?
+            # https://github.com/pytorch/pytorch/issues/57459
+            return F.kl_div(
+                ref_logprob, logprob, log_target=True, reduction="none"
+            ).sum(-1)
```

Comment:
Why sum and not mean here?

Reply:
I've another question `ref_logprob, logprob` order. Why is it this order i.e. 
```
input = ref_logprob
target = logprob
```

Reply:
Good question about the sum. I took it from the trl implementation. I'd need to check, but this should not be the token dimension that is being reduced. 

https://github.com/pytorch/pytorch/issues/57459 it seems to be inverting the positional arguments. Apparently this way is correct, as the KL_div is positive.

Reply:
Thanks, I think it should be good then